### DISCUSSION TITLE
Query

### DISCUSSION THREAD

**Discussion by mujtaba-fleapo** [chatscope/use-chat#26] [Q&A]
Hi, kind of a stupid question but I need to know the difference between the **sendMessage** and **addMessage** functions provided by use-chat, and what can be the use cases for each.
Again, I apologize if this question seems too naive.
Thank you!

**Comment by supersnager** [comment]
@mujtaba-fleapo ```sendMessage``` is used to sending message to the backend e.g. using websocket, ```add message``` is used to adding message object to collection of messages that are available under
```typescript
const messages = useChat();
```

```sendMessage``` internally ads the message to the store before calling ```sendMessage``` method of the provider: https://github.com/chatscope/use-chat/blob/43a239c973119acece801fe1d523115328521744/src/provider/ChatProvider.tsx#L403

So, if you send a message it will be immediately available in the useChat hook.
